src/main/java/com/mraof/minestuck/api/alchemy/GristType.java

```diff
@@ -0,0 +1,205 @@
+	
+	public GristType(Properties properties)
+	{
+		rarity = properties.rarity;
+		value = properties.value;
+		candyItem = properties.candyItem;
+		underlingData = properties.underlingData;
+		textureOverrideId = properties.textureOverrideId;
+	}
+	
+	@Nullable
+	public ResourceLocation getId()
+	{
+		return GristTypes.getRegistry().getKey(this);
+	}
+	
+	public ResourceLocation getIdOrThrow()
+	{
+		return Objects.requireNonNull(this.getId());
+	}
+	
+	public Component getNameWithSuffix()
+	{
+		return Component.translatable(FORMAT, getDisplayName());
+	}
+	
+	public MutableComponent getDisplayName()
+	{
+		return Component.translatable(getTranslationKey());
+	}
+	
+	public String getTranslationKey()
+	{
+		return translationKey.get();
+	}
+	
+	/**
+	 * Returns the grist's rarity. Is a number from 0.0 to 1.0.
+	 */
+	public float getRarity()
+	{
+		return rarity;
+	}
+	
+	/**
+	 * Returns a modifier used for the power of underlings of the grist's type.
+	 */
+	public float getPower()
+	{
+		return underlingData != null ? underlingData.power : 0;
+	}
+	
+	/**
+	 * @return a value estimate for this grist type
+	 */
+	public float getValue()
+	{
+		return value;
+	}
+	
+	public boolean isUnderlingType()
+	{
+		return underlingData != null;
+	}
+	
+	public ResourceLocation getIcon()
+	{
+		return icon.get();
+	}
+	
+	public ResourceLocation getTextureId()
+	{
+		if(this.textureOverrideId != null)
+			return this.textureOverrideId;
+		
+		return Objects.requireNonNullElse(this.getId(), DUMMY_ID);
+	}
+	
+	public ItemStack getCandyItem()
+	{
+		return candyItem.get();
+	}
+	
+	public List<GristType> getSecondaryTypes()
+	{
+		return Objects.requireNonNull(GristTypes.getRegistry().tags())
+				.getTag(this.getSecondaryTypesTag()).stream().toList();
+	}
+	
+	public TagKey<GristType> getSecondaryTypesTag()
+	{
+		return TagKey.create(GristTypes.REGISTRY_KEY, this.getIdOrThrow().withPrefix("secondary/"));
+	}
+	
+	public int getColor()
+	{
+		return underlingData != null ? underlingData.color : 0xFFFFFF;
+	}
+	
+	public static class Properties
+	{
+		private final float rarity, value;
+		private Supplier<ItemStack> candyItem = () -> ItemStack.EMPTY;
+		@Nullable
+		private UnderlingData underlingData = null;
+		@Nullable
+		private ResourceLocation textureOverrideId = null;
+		
+		public Properties(float rarity, float value)
+		{
+			this.rarity = rarity;
+			this.value = value;
+		}
+		
+		public Properties candy(Supplier<Item> item)
+		{
+			Objects.requireNonNull(item);
+			return candyStack(() -> new ItemStack(item.get()));
+		}
+		
+		public Properties candyStack(Supplier<ItemStack> stack)
+		{
+			candyItem = Objects.requireNonNull(stack);
+			return this;
+		}
+		
+		public Properties underlingType(int color)
+		{
+			return this.underlingType(1 / this.rarity, color);
+		}
+		
+		public Properties underlingType(float power, int color)
+		{
+			this.underlingData = new UnderlingData(color, power);
+			return this;
+		}
```

Comment:
This seems to replace the method color, but without context or a description it seems kind of confusing

Reply:
Would the problem be solved with some javadoc saying that the color parameter is used to color the texture for underlings of that type?

Reply:
I think so!